cmd/dockerd/docker.go

```diff
@@ -23,6 +25,16 @@ func main() {
 		return
 	}
 
+	// Fixes https://github.com/docker/docker/issues/19728
+	// Ignore SIGPIPE events, that are generated by systemd when journald is restarted while
+	// the docker daemon is not restarted and also running under systemd.
+	go func() {
+		c := make(chan os.Signal)
+		signal.Notify(c, os.Signal(syscall.SIGPIPE))
+		for range c {
```

Comment:
Just to be clear if you log a message in this handler:

Docker catches the SIGPIPE, would log to stdout/stderr, systemd would generate a SIGPIPE event, and the whole cycle repeats.